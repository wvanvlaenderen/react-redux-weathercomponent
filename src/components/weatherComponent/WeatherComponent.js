import React from 'react'
import './style.css'

export default class WeatherComponent extends React.Component {

  kelvinToCelcius (temperature) {
    return Math.round(temperature - 273.15)
  }

  render () {
    const {weather} = this.props

    return (
        <div className="weather-card">
          <div className="weather-icon cloud"></div>
          <h1 id="temp" className="temp">
            {this.kelvinToCelcius(weather.temp)}ยบ
          </h1>
          <br />
          <p id="location">{weather.location}</p>
        </div>
    )
  }
}