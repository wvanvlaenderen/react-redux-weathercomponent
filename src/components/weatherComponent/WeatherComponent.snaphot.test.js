import React from 'react'
import renderer from 'react-test-renderer'
import WeatherComponent from './WeatherComponent'

test('Renders temperature in kelvin to celcius using snaphot testing', () => {
  const component = renderer.create(
      <WeatherComponent
          weather={{
            location: 'london',
            temp: 300,
          }}
      />
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})