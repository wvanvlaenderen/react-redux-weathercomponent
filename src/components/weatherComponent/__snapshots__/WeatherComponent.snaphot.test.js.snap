// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders temperature in kelvin to celcius using snaphot testing 1`] = `
<div
  className="weather-card"
>
  <div
    className="weather-icon cloud"
  />
  <h1
    className="temp"
    id="temp"
  >
    27
    ยบ
  </h1>
  <br />
  <p
    id="location"
  >
    london
  </p>
</div>
`;
